synapse/pipeline.yml

```diff
@@ -323,9 +308,12 @@ steps:
   # We mostly care about testing each topology.
   # To vary the Python and Postgres versions, we use Docker images which are based
   # on an assortment of Debian releases.
-  #   - "py35" is Debian  9 Stretch (Python 3.5, Postgres 9.6)
-  #   - "py37" is Debian 10 Buster (Python 3.7, Postgres 11)
-  #   - "py38" is Debian 11 Bullseye (Python 3.9, Postgres 13)
+  #   - "bionic" (Ubuntu 18.04) has Python 3.6 and Postgres 10
+  #   - "buster" (Debian 10) has Python 3.7 and Postgres 11
+  #   - "bullseye" (Debian 11) (currently) has Python 3.9 and Postgres 13
+  #
+  # TODO: this leaves us without sytests for Postgres 9.6. How much do we care
+  #    about that?
```

Comment:
It's quite unfortunate, really. I wonder if we should be using docker compose here to allow us to more accurately select python and postgres versions? That's not really a small change though....

Reply:
This was discussed in #synapse-dev and the rough conclusion was that Pg9.6 goes EOL in November (only 7 months away! Eek!) so it's not worth addressing immediately since we do get coverage of the very next version by targeting Bionic.

And some time between now and November we'll do the proper thing and split out Python / Postgres versions. My hope is that if GitHub Actions proves fruitful making this sort of change will be (a) easier, and (b) accessible to the community (Jonathan has already volunteered).